src/api/contractsService.ts

```diff
@@ -0,0 +1,17 @@
+import { ContractsType } from '../components/shared/types/contracts'
+import client from './axios'
+
+const baseUrl = '/contract'
+
+const getAll = async (): Promise<ContractsType[]> => {
+  try {
+    const result = await client.get(baseUrl)
+    return result.data
+  } catch (error) {
+    throw new Error('Contratos não encontrados')
```

Comment:
O nest ja tem algumas exceçoes nativas pra usar, da uma olhada em https://docs.nestjs.com/exception-filters#built-in-http-exceptions

Reply:
não entendi muito bem como utilizar as exceptions do nest no react, em vez disso eu deixei a responsabilidade de tratar o erro com o chamador da função que fez a requisição dos dados, então o component que fica responsável por mostrar notificação de erro caso seja necessário

Reply:
A ideia era implementar as exceptions da mesma forma que o nest, nao seria usar as excecoes do nest no react. Vc pode implementar um set de exceptions igual o nest faz e usar na aplicacao. podemos fazer isso em outro PR.